### PR TITLE
Implement session cache using Redis

### PR DESCRIPTION
**Pull request by sebastian-rivas**
# Description
- Default (and only) implementation of Nakama's session cache uses in-memory cache so in server restart the sessions are lost (even refresh token won't work anymore). This implements session cache using Redis.

### CODE REVIEW

**Review by fcurti-ar — APPROVED** [review]

File: server/config.go [hunk 1]
```diff
@@ -570,6 +572,10 @@ func (c *config) GetIAP() *IAPConfig {
 	return c.IAP
 }
 
+func (c *config) GetCache() *CacheConfig {
```
Comment (srebolledo) [inline server/config.go:576]: method not commented

**Review by srebolledo — APPROVED** [review]
Minor comment :)

File: server/session_cache_redis.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	cachedUser := s.getCachedUser(userID)
+	if cachedUser == nil {
+		return false
+	}
+
+	return cachedUser.RefreshToken == token && cachedUser.RefreshExp >= time.Now().UTC().Unix()
+}
+
+// Add is used when a user should be added to session cache.
+func (s *RedisSessionCache) Add(userID uuid.UUID, sessionExp int64, sessionToken string, refreshExp int64, refreshToken string) {
```
Comment (josh-english-2k18) [inline server/session_cache_redis.go]: Probably a good idea to add a suffix to `refreshExp` describing what type of time value it is - millis, seconds, etc.

Ditto with internal time variables

**Review by josh-english-2k18 — APPROVED** [review]
Overall looks good, please address comments before merge.

File: server/session_cache_redis.go (hunk 2 above)
Comment (sebastian-rivas) [reply server/session_cache_redis.go]: well yes, I was just trying to keep the signature but I'll do the rename

**Pull request merged**